core/camel-core-model/src/main/java/org/apache/camel/model/RoutesDefinition.java

```diff
@@ -192,9 +192,18 @@ public RouteDefinition from(EndpointConsumerBuilder endpoint) {
      * @return       the builder
      */
     public RouteDefinition route(RouteDefinition route) {
+        // must set the error handler if not already set on the route
+        if (route.getErrorHandlerFactory() != null && route.getErrorHandlerRef() != null) {
+            ErrorHandlerFactory handler = getErrorHandlerFactory();
+            if (handler != null) {
+                route.setErrorHandlerFactory(handler);
+            }
+        }
```

Comment:
@davsclaus I'm not sure if this is actually correct but if a route is added using a `RouteDefinition`, the error handler set on the route collection is not propagated to the route which leads to different behavior depending to which method you'd use. This is probably not supposed to be used by developers but it is needed for my YAML DSL work.

Let me know if there is a better way.

Reply:
Ah yeah that API is not really for end users, as that would mean they build those model classes outside the DSL and then take that model (which is a route) and add to a route (doesnt make so much sense).

We should possibly mark in the javadoc that this is only for internal usage. And maybe come up with a different name, and deprecate the old name.

Reply:
I've opened an [issue](https://issues.apache.org/jira/browse/CAMEL-16211) to track the renaming and documentation task.

Anything else that need to be addressed before merging ?

Reply:
No it looks good